SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenFrames.swift

```diff
@@ -0,0 +1,107 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import QuartzCore
+import UIKit
+
+// swiftlint:disable shorthand_operator
+class ScreenFrames: NSObject {
+
+    private var displayLink: CADisplayLink?
+
+    private var slowFrameThreshold: CFTimeInterval = SplunkRum.configuredOptions?.slowFrameThreshold ?? 16.7
+    private var frozenFrameThreshold: CFTimeInterval = SplunkRum.configuredOptions?.frozenFrameThreshold ?? 700
+
+    private var currentIteration: Int = 0
+    private var startedTime: CFTimeInterval = CACurrentMediaTime()
+
+    private var slowCount: Int = 0
+    private var frozenCount: Int = 0
+    private var isFirstIteration: Bool = true
+    private var previousTimestamp: CFTimeInterval = CACurrentMediaTime()
+
+    func startTracking() {
+        frozenFrameThreshold = frozenFrameThreshold / 1000
+        slowFrameThreshold = slowFrameThreshold / 1000
+        stopTracking() /// make sure to stop a previous running display link
+        let displayLink = CADisplayLink(target: self, selector: #selector(displayLinkCallback))
+        displayLink.add(to: .main, forMode: .common)
+        self.displayLink = displayLink
+    }
+
+    func stopTracking() {
+
+        displayLink?.invalidate()
+        displayLink = nil
+    }
+
+    // timestamp: The time interval that represents when the last frame displayed
+    // targetTimestamp: TargetTimestamp is the time of the next frame to trigger the link
+    @objc func displayLinkCallback(_ displayLink: CADisplayLink) {
+
+        // In order to manage the long render time, after bringing the app back into the foreground.
+         let state = UIApplication.shared.applicationState
+         if state == .background || state == .inactive {
+            isFirstIteration = true
+         }
+         let currentTime: CFTimeInterval = CACurrentMediaTime()
+         if isFirstIteration {
+            previousTimestamp = currentTime
+            isFirstIteration = false
+         }
+        // Report every slow frame as a span, in 1 second intervals, in which the count of slow frames is recorded.
+         let duration = displayLink.timestamp - previousTimestamp
+         previousTimestamp = displayLink.timestamp
+         let elapsedTime = currentTime - startedTime
+         let iteration = Int(elapsedTime)
+
+         if duration > frozenFrameThreshold {
+            frozenCount += 1
+         } else if duration > slowFrameThreshold {
+            slowCount += 1
+         }
+
+         if currentIteration != iteration {
+
+             if slowCount > 0 {
+                 reportSlowframe(slowFrameCount: slowCount, name: "slowRenders")
+             }
+
+             if frozenCount > 0 {
+                 reportSlowframe(slowFrameCount: frozenCount, name: "frozenRenders")
```

Comment:
The reporting happens on the next second. What if the count was increased during the previous second while a different screen was active? The screen name will be reported wrongly?

Reply:
Consider this scenario:

```
time=0.5s | User is on screen A, there is a slow frame, slowCount = 1
time=0.6s | User moves to screen B
time=1.0s | currentItartion != iteration part is now true -> reportSlowFrame uses `getScreenName` which returns screen B
```

Reply:
Solved issue with screen name reported incorrectly.